File: topic_3/Haskell/3-iguais.hs
-- Retorne quantos números dos 3 são iguais
iguais a b c
  | a == b && b == c = 3
  | a == b || a == c || b == c = 2
  | otherwise = 0

main = do
    a <- readLn :: IO Int
    b <- readLn :: IO Int
    c <- readLn :: IO Int
    print $ iguais a b c

File: topic_3/Python/3-iguais.py
# Retorne quantos números dos 3 são iguais

def iguais(a, b, c):
    cont = 0
    if(a == b and b == c):
        return 3
    elif(a == b or a == c or b == c):
        return 2
    else:
        return 0

print(iguais (2, 2, 2) == 3) #True
print(iguais (2, 2, 3) == 2) #True
print(iguais (2, 3, 2) == 2) #True
print(iguais (2, 1, 1) == 2) #True
print(iguais (3, 2, 1) == 0) #True